src/test/java/org/sagebionetworks/bridge/services/AccountServiceTest.java

```diff
@@ -1,167 +1,954 @@
     @Test
     public void createAccount() {
         Study study = Study.create();
+        study.setIdentifier(TEST_STUDY_IDENTIFIER);
+        
         Account account = Account.create();
+        account.setId(USER_ID);
+        account.setEmail(EMAIL);
+        account.setStatus(UNVERIFIED);
+        account.setStatus(ENABLED);
+        account.setStudyId("wrong-study");
         Consumer<Account> consumer = (oneAccount) -> {};
+
         service.createAccount(study, account, consumer);
-        verify(mockAccountDao).createAccount(study, account, consumer);
+        verify(mockAccountDao).createAccount(eq(study), accountCaptor.capture(), eq(consumer));
+        
+        Account createdAccount = accountCaptor.getValue();
+        assertEquals(createdAccount.getId(), USER_ID);
+        assertEquals(createdAccount.getStudyId(), TEST_STUDY_IDENTIFIER);
+        assertEquals(createdAccount.getCreatedOn().getMillis(), MOCK_DATETIME.getMillis());
+        assertEquals(createdAccount.getModifiedOn().getMillis(), MOCK_DATETIME.getMillis());
+        assertEquals(createdAccount.getPasswordModifiedOn().getMillis(), MOCK_DATETIME.getMillis());
+        assertEquals(createdAccount.getStatus(), ENABLED);
+        assertEquals(createdAccount.getMigrationVersion(), MIGRATION_VERSION);
     }
-    
+
     @Test
-    public void updateAccount() {
-        Account account = Account.create();
+    public void updateAccount() throws Exception {
+        Account account = mockGetAccountById(ACCOUNT_ID, false);
         Consumer<Account> consumer = (oneAccount) -> {};
+
         service.updateAccount(account, consumer);
         verify(mockAccountDao).updateAccount(account, consumer);
     }
     
     @Test
-    public void editAccount() {
-        Consumer<Account> consumer = (oneAccount) -> {};
-        service.editAccount(TEST_STUDY, HEALTH_CODE, consumer);
-        verify(mockAccountDao).editAccount(TEST_STUDY, HEALTH_CODE, consumer);
-    }
-    
-    @Test
-    public void getAccount() {
+    public void updateAccountNotFound() {
+        // mock hibernate
         Account account = Account.create();
-        when(mockAccountDao.getAccount(ACCOUNT_ID)).thenReturn(account);
+        account.setStudyId(TEST_STUDY_IDENTIFIER);
+        account.setId(USER_ID);
+        when(mockAccountDao.getAccount(any())).thenReturn(Optional.empty());
         
+        // execute
+        try {
+            service.updateAccount(account, null);
+            fail("expected exception");
+        } catch (EntityNotFoundException ex) {
+            assertEquals(ex.getMessage(), "Account not found.");
+        }
+    }
+
+    @Test
+    public void editAccount() throws Exception {
+        AccountId accountId = AccountId.forHealthCode(TEST_STUDY_IDENTIFIER, HEALTH_CODE);
+        Account account = mockGetAccountById(accountId, false);
+
+        service.editAccount(TEST_STUDY, HEALTH_CODE, mockConsumer);
+
+        verify(mockAccountDao).updateAccount(account, mockConsumer);
```

Comment:
Need to verify that the consumer is called before update.

Reply:
Added